aesd-char-driver/main.c

```diff
@@ -0,0 +1,295 @@
+/**
+ * @file aesdchar.c
+ * @brief Functions and data related to the AESD char driver implementation
+ *
+ * Based on the implementation of the "scull" device driver, found in
+ * Linux Device Drivers example code.
+ *
+ * @author Dan Walkes
+ * @date 2019-10-22
+ * @copyright Copyright (c) 2019
+ *
+ */
+
+#include <linux/module.h>
+#include <linux/init.h>
+#include <linux/printk.h>
+#include <linux/types.h>
+#include <linux/cdev.h>
+#include <linux/fs.h> // file_operations
+#include "aesdchar.h"
+#include <linux/syscalls.h>
+#include <asm/uaccess.h>
+
+int aesd_major =   0; // use dynamic major
+int aesd_minor =   0;
+
+MODULE_AUTHOR("Pranav Bharadwaj"); /** TODO: fill in your name **/
+MODULE_LICENSE("Dual BSD/GPL");
+
+struct aesd_dev aesd_device;
+
+int aesd_open(struct inode *inode, struct file *filp)
+{
+    PDEBUG("open");
+    /**
+     * TODO: handle open
+     */
+    //Increment device use count by a process
+    try_module_get(THIS_MODULE);
```

Comment:
@prvbharadwaj1731 I'm curious where you found this, I don't think it's necessary.  See https://stackoverflow.com/a/6079839